image/helpers.go

```diff
@@ -404,3 +406,56 @@ func (tsto *ToolingStore) Find(at *asserts.AssertionType, headers map[string]str
 	}
 	return tsto.sto.Assertion(at, pk, tsto.user)
 }
+
+// var so that it can be mocked for tests
+var writeResolvedContent = writeResolvedContentImpl
+
+// XXX: move to gadget?
+func writeResolvedContentImpl(prepareDir, gadgetUnpackDir, kernelUnpackDir string, model *asserts.Model) error {
+	// XXX: ugly this is hardcoded here and that the caller needs
+	// to know what we did here :/
```

Comment:
this sentence is hard to parse, did you mean to say something like

```suggestion
	// XXX: this is ugly; we hardcode this path here because the caller needs
	// to know what we did here :/
```

Reply:
I removed this now and tweaked the signature of the function